### PR TITLE
Fix to support CV 18

### PR DESCRIPTION
**Pull request by cvl01**
## What does this PR do?

Currently `recipes/CommonVoice/common_voice_prepare.py` throws an error when used with Common Voice 18. Seems that the columns used have been changed, a column was added. 

This solution tries to be more bullet proof by using the header row and mapping that, to determine the column index. 

Fixes # (no issue)

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by TParcollet** [comment]
@asumagic we were talking about it! This sounds like a nice update thanks @cvl01

File: recipes/CommonVoice/common_voice_prepare.py [hunk 1]
```diff
@@ -309,8 +314,14 @@ def create_csv(
...
-    loaded_csv = open(orig_tsv_file, "r").readlines()[1:]
-    nb_samples = len(loaded_csv)
+    csv_lines = open(orig_tsv_file, "r").readlines()
+    header_line = csv_lines[0]
+    csv_data_lines = csv_lines[1:]
+    nb_samples = len(csv_data_lines)
+
+    header_map = {
+        column_name: index for index, column_name in enumerate(header_line)
+    }
```
Comment (pplantinga) [inline recipes/CommonVoice/common_voice_prepare.py:324]: Doesn't `csv.DictReader` handle all of this more elegantly?

Comment (asumagic) [reply recipes/CommonVoice/common_voice_prepare.py:324]: As discussed yesterday, will review/test as is and not block on this

**Comment by asumagic** [comment]
After my fixes, sanity checked on sr 19.0, ka 6.1, en (of whatever unlabelled version we have on this cluster?) and the output CSVs seem correct. CV en training on conformer-transducer runs.

**Review by asumagic — APPROVED** [review]

**Pull request merged**